### PR TITLE
Add Catch unit testing capability and some basic tests for the surrogate T/H driver

### PR DESCRIPTION
**Pull request by aprilnovak**
This MR adds unit test capabilities with the [Catch2](https://github.com/catchorg/Catch2) unit testing framework. This is a header-only unit testing framework, so I just added it to the `vendor` directory - if there is a different preferred way of doing this let me know!

I also added some simple unit tests for reading the input parameters of the `SurrogateHeatDriver`. These tests can be run by building the `unittests` executable and running it from the `enrico/unit` directory.

Basically, I'm at the point where I want to have some unit tests for all of the data mappings going on to keep my head straight. So whether it's Catch2 or something else, I think it's time! Follow-on MRs will add some more robust tests for the `SurrogateHeatDriver` and for its mappings with OpenMC models.

This doesn't set up any type of PR-initiated unit tests, which is something we should add in the future.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -148,3 +148,13 @@ if (SCALE_FOUND)
     test_shift_singlerod
     PROPERTIES CXX_STANDARD 14 CXX_EXTENSIONS OFF)
 endif ()
+
+# =============================================================================
+# Build unit tests
+# =============================================================================
+add_library(Catch INTERFACE)
+target_include_directories(Catch INTERFACE src/vendor/catch)
```
Comment (paulromano) [inline CMakeLists.txt]:
```suggestion
target_include_directories(Catch INTERFACE vendor/catch)
```

File: unit/catch.cpp [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#define CATCH_CONFIG_MAIN
+#include "catch/catch.hpp"
```
Comment (paulromano) [inline unit/catch.cpp]: If `vendor/catch` is given as the include directory, this should just be:
```suggestion
#include "catch.hpp"
```

File: unit/test_surrogate_th.cpp [hunk 3]
```diff
@@ -0,0 +1,432 @@
+/**
+ * \file test_surrogate_th.cpp
+ * \brief Unit tests for surrogate T/H solver.
+ */
+
+#include "catch/catch.hpp"
```
Comment (paulromano) [inline unit/test_surrogate_th.cpp]:
```suggestion
#include "catch.hpp"
```

**Comment by paulromano** [comment]
@aprilnovak One other request -- since we are using git submodules for the other third-party dependencies, we might as well do that for Catch2 as well. Can you switch to using a git submodule instead of including the file directly?

**Comment by aprilnovak** [comment]
Sure, I can make this a submodule. And @RonRahaman - just clarifying if you're okay with the current directory location or if you'd like it in `tests/surrogate_th`? I see two comments but only one says disregard

**Comment by RonRahaman** [comment]
@aprilnovak I do want to change the location, but it turns out I don't know how to converse on GitHub, and I keep changing my mind.  

I'm certain that I'd like the unit tests to be inside `tests` somehow.  The rest depends on what your goals are.  Here are the ideas that occur to me:

* Do you want to have one unit test directory that contains all the test drivers?  In which case, it seems reasonable to have a `tests/unit/` directory that contains `test_surrogate_th.cpp` along with whatever future unit tests we have (for example, `test_nek_driver.cpp` or whatever).  
* Do you want to have a separate directory for each test driver?  In this case, it seems reasonable to have `tests/unit/surrogate_th/` now and, in the future, `tests/unit/nek_driver/` etc.


But as long as your new drivers are inside `tests/`, it's totally up to you.

**Comment by aprilnovak** [comment]
@rahaman, that sounds good to me! I think I'd like it in `tests/unit` because I plan on having a test for multiple drivers that shares the same set of XML files. If it gets too unwieldy in the future we can separate the tests into multiple directories, too.
